### DISCUSSION TITLE
Starlite Tutorial

### DISCUSSION THREAD

**Discussion by JacobCoffee** [litestar-org/litestar#1158] [Ideas]
**Prompt**: Prospective framework users tend to do a few things when evaluating projects for their use case. Once the user clones the repo and begins to launch a demo app, they may want some ELI5 style instructions to help them on their way.

The following proposal is to generate a tutorial page within existing documentation that can do any of the following:

- Guide users into building a "toy" app in a tutorial
- Provide step-by-step instructions, beginning with very basic and expanding into other more advanced topics

Some examples of extremely well done guides:
- [Django](https://docs.djangoproject.com/en/4.1/intro/tutorial01/)
- [Flask](https://flask.palletsprojects.com/en/2.2.x/tutorial/)

Some examples of other frameworks that do a great job:
- [Falcon](https://falcon.readthedocs.io/en/stable/user/tutorial.html#)

Some examples of not so clear but still good guides:
- [FastAPI](https://fastapi.tiangolo.com/tutorial/)

Some examples of somewhat hard to follow guides:
- [Starlite](https://starlite-api.github.io/starlite/latest/usage/the-starlite-app.html)
- [Sanic](https://sanic.dev/en/guide/basics/app.html#instance)

I hope this paints a clear picture on what the proposal is after here. None of these are bad, it is just in the interest of this proposal to craft a guide whereby users may confidently stand up their projects and progress from beginner to advanced in stages.

**Comment by provinzkraut** [comment]
Already working on it. It will be a Django-style tutorial like the one you linked.

**Reply by JacobCoffee** [reply]
https://github.com/orgs/starlite-api/projects/1/views/1?pane=issue&itemId=16646761
